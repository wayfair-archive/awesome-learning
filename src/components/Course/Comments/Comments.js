import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import ReactDisqusComments from 'react-disqus-comments';

export const PureComments = ({ data, courseTitle, courseSlug }) => {
  const {
    url,
    disqusShortname
  } = data.site.siteMetadata;

  if (!disqusShortname) {
    return null;
  }

  return (
    <ReactDisqusComments
      shortname={disqusShortname}
      identifier={courseTitle}
      title={courseTitle}
      url={url + courseSlug}
    />
  );
};

export const Comments = (props) => (
  <StaticQuery
    query={graphql`
      query CommentsQuery {
        site {
          siteMetadata {
            disqusShortname
            url
          }
        }
      }
    `}
    render={(data) => <PureComments {...props} data={data}/>}
  />
);

export default Comments;
