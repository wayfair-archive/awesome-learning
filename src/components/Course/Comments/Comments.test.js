import React from 'react';
import renderer from 'react-test-renderer';
import { PureComments as Comments } from './Comments';

describe('Comments', () => {
  it('renders correctly', () => {
    const props = {
      data: {
        site: {
          siteMetadata: {
            url: 'http://localhost',
            disqusShortname: 'test'
          }
        }
      },
      courseTitle: 'test',
      courseSlug: '/test'
    };

    const tree = renderer.create(<Comments {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
