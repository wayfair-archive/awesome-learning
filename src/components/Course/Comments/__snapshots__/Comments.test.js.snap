// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders correctly 1`] = `
<div>
  <div
    id="disqus_thread"
  />
</div>
`;
